client/src/components/RecipeList/components/RecipeCard.jsx

```diff
@@ -1,42 +1,195 @@
+} from 'antd';
+import {
+  InfoCircleOutlined, StarOutlined, ExportOutlined, CloseCircleOutlined, StarFilled,
+} from '@ant-design/icons';
+import RecipeNames from './RecipeConsts';
 
 const { Meta } = Card;
 
-const openRecipe = (link) => {
-  window.open(link, 'noopener noreferrer');
-};
+class RecipeCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { showModal: false, viewMore: false };
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
+  }
 
-const RecipeCard = (props) => {
-  const { recipe } = props;
-  const { title, link, image: imageURL } = recipe;
-  return (
-    <Card
-      className="recipe-card"
-      onClick={() => openRecipe(link)}
-      hoverable
-      cover={(
-        <img
-          className="recipe-image"
-          src={imageURL}
-          alt={title}
-        />
-      )}
-    >
-      <Meta
-        className="recipe-card-meta"
-        title={title}
-      />
-    </Card>
-  );
-};
+
+  showModal() {
+    this.setState({
+      showModal: true,
+    });
+  }
+
+  hideModal() {
+    this.setState({
+      showModal: false,
+    });
+  }
+
+
+  render() {
+    const { recipe } = this.props;
+    const {
+      title, link, image: imageURL, total_time: totalTime, rating, number_of_raters: numberOfRaters,
+      tags, id,
+    } = recipe;
+    const viewMoreText = `${tags.length - 5} more`;
+    const { showModal, viewMore } = this.state;
+
+    const openRecipe = () => {
+      window.open(link, 'noopener noreferrer');
+    };
+
+    const findSite = () => Object.entries(RecipeNames)
+      .find((pair) => new RegExp(pair[0]).test(id))[1];
+
+    const site = findSite();
+
+    return (
+      <>
+        <Card
+          className="recipe-card"
+          // onClick={() => openRecipe(link)}
+          hoverable
+          cover={(
+            <img
+              className="recipe-image"
+              src={imageURL}
+              alt={title}
+            />
+          )}
+
+          actions={[
+            <Button className="recipe-button" block type="link" size="large" onClick={this.showModal}>
+              <InfoCircleOutlined />
+            </Button>,
+            <Button className="recipe-button" block type="link" size="large">
+              <StarOutlined />
+            </Button>,
+            <Button className="recipe-button" block type="link" size="large" onClick={openRecipe}>
+              <ExportOutlined />
+            </Button>,
+          ]}
+        >
+          <Meta
+            className="recipe-card-meta"
+            title={title}
+            description={site}
+          />
+        </Card>
+        <Modal
+          title={title}
+          visible={showModal}
+          onCancel={this.hideModal}
+          footer={[
+            <Button key="Save">
+              <StarOutlined />
+            </Button>,
+            <Button key="Go" onClick={openRecipe}>
+              <ExportOutlined />
```

Comment:
I suggest that we add some text to the button in the modal, or a popover (also in ant) explaining which is which.